File: TranslationFiestaFSharp/ProviderIds.fs
namespace TranslationFiestaFSharp

module ProviderIds =
    let GoogleUnofficial = "google_unofficial"

    let normalize (value: string | null) =
        let trimmed =
            match value with
            | null -> ""
            | text -> text.Trim().ToLowerInvariant()
        match trimmed with
        | "unofficial" -> GoogleUnofficial
        | "google_unofficial_free" -> GoogleUnofficial
        | "google_free" -> GoogleUnofficial
        | "googletranslate" -> GoogleUnofficial
        | "" -> GoogleUnofficial
        | _ -> GoogleUnofficial

File: TranslationFiestaCSharp/ProviderIds.cs
using System;

namespace TranslationFiestaCSharp
{
    public static class ProviderIds
    {
        public const string GoogleUnofficial = "google_unofficial";

        public static string Normalize(string? value)
        {
            var normalized = (value ?? string.Empty).Trim().ToLowerInvariant();
            return normalized switch
            {
                "unofficial" => GoogleUnofficial,
                "google_unofficial_free" => GoogleUnofficial,
                "google_free" => GoogleUnofficial,
                "googletranslate" => GoogleUnofficial,
                "" => GoogleUnofficial,
                _ => GoogleUnofficial
            };
        }
    }
}

